### PR TITLE
docs(react-persona): Adding Persona spec

### PR DESCRIPTION
**Pull request by sopranopillow**
This is the initial spec, I would love some feedback!

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/cde2f43f6bf0522ddf189dbf3b2905ec1f2f7f88'>cde2f43f6bf0522ddf189dbf3b2905ec1f2f7f88</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-persona/Spec.md [hunk 1]
```diff
@@ -1,63 +1,272 @@
...
 
-_Include background research done for this component_
+- OpenUI research: [Avatar/Persona](https://open-ui.org/components/avatar.research)
+- v7/v8: [link](https://developer.microsoft.com/en-us/fluentui#/controls/web/persona)
+  - > Note: v8's Persona is a combination of v9's Persona and Avatar (v8's Persona has a sub-component PersonaCoin that would count as Avatar, but is not exposed.). In v9 we've taken the approach of having a single component Avatar and another component Persona.
+- v0: Does not have equivalent component.
+
+### Coin
+
+_Avatar vs PersonaCoin_: As mentioned in Prior Art, v8's persona is not quite the same as v9's. v8's Persona is only able to showcase PersonaCoin while v9 showcases Avatar (including Avatar + PresenceBadge, Avatar + image, and Avatar + Icon).
```
Comment (GeoffCoxMSFT) [inline packages/react-components/react-persona/Spec.md:20]: This is the first place you reference PersonaCoin.  It would be helpful to anyone reading that isn't familiar with the differences between Avatar and PersonaCoin to have a visual to show their differences (or description of the visual differences).

File: packages/react-components/react-persona/Spec.md [hunk 2]
```diff
@@ -1,63 +1,272 @@
 # @fluentui/react-persona Spec
 
+Convergence epic issue: #24213
+
 ## Background
 
-_Description and use cases of this component_
+A Persona is a visual representation that showcases an Avatar, Icon, Image, or Badge.
+
+Persona is used in PeoplePicker, Team's left rail and menus, chiclets, and Card.
```
Comment (GeoffCoxMSFT) [inline packages/react-components/react-persona/Spec.md:9]: I think it is worth calling out that this is a composite component but is not a card in itself - it won't be providing a visual board or popup when the user hovers over an avatar.

File: packages/react-components/react-persona/Spec.md [hunk 3]
```diff
@@ -1,63 +1,279 @@
...
-- _Link to GitHub epic issue for the converged component_
+- Note: The icon only shows up if there's no presence and size is `tiny`, `size8`, or `size16`.
+
+_Image vs PersonaCoin_: PersonaCoin allows you to have an image only within the Avatar. This can be avoided by rendering a custom coin, but there's still all the dom and styling overhead that comes with PersonaCoin. v9's Persona allows you use a single image with no extra dom.
+
+_PresenceBadge vs PersonaCoin_: PersonaCoin allows you to have only a PresenceBadge when the size of the coin is `tiny`, `size8`, or `size16`. In v9's Persona you are able to render the PresenceBadge alone.
+
+### Text lines
+
+Other than styling and naming, the text lines in Persona remain the same.
```
Comment (layershifter) [inline packages/react-components/react-persona/Spec.md:40]: It would be great to include a section with a component anatomy as it starts with sample code, but we don't know how it looks 🤔

File: packages/react-components/react-persona/Spec.md [hunk 4]
```diff
@@ -1,63 +1,279 @@
...
+
+Persona with Avatar + PresenceBadge:
+
+```jsx
+<Persona
+  primaryText="Kevin Sturgis"
+  secondaryText="Software Engineer"
+  tertiaryText="Offline"
+  avatar={{ name: 'Kevin Sturgis' }}
+  badge={{ status: 'offline', outOfOffice: true }}
```
Comment (layershifter) [inline packages/react-components/react-persona/Spec.md:57]: Should `badge` be in `avatar` slot?

```suggestion
  avatar={{ name: 'Kevin Sturgis', badge: { status: 'offline', outOfOffice: true } }}
```

File: packages/react-components/react-persona/Spec.md [hunk 5]
```diff
@@ -1,63 +1,279 @@
...
 ## Variants
 
-_Describe visual or functional variants of this control, if applicable. For example, a slider could have a 2D variant._
+There are three alignment variants:
+
+- start: Coin on the left and text on the right.
+- center: Coin on top and text on the bottom.
+- end: Coin on the right and text on the left.
+
+There are 5 coin variats:
```
Comment (layershifter) [inline packages/react-components/react-persona/Spec.md:93]:
```suggestion
There are 5 coin variants:
```

File: packages/react-components/react-persona/Spec.md [hunk 6]
```diff
@@ -1,63 +1,279 @@
...
+```
 
 ## Variants
 
-_Describe visual or functional variants of this control, if applicable. For example, a slider could have a 2D variant._
+There are three alignment variants:
+
+- start: Coin on the left and text on the right.
+- center: Coin on top and text on the bottom.
+- end: Coin on the right and text on the left.
```
Comment (layershifter) [inline packages/react-components/react-persona/Spec.md:91]: I suggest to use "media slot" or "media" as for me "coin" means a special shape and is more opinionated about its content.

File: packages/react-components/react-persona/Spec.md [hunk 7]
```diff
@@ -1,63 +1,279 @@
...
+
+There are 5 coin variats:
+
+- Badge
+- icon
+- Avatar
+- Avatar + Badge
+  - > Note: When there's an Avatar and a Badge and the Avatar provided already has a Badge, the provided Badge will take over the provided Badge.
```
Comment (layershifter) [inline packages/react-components/react-persona/Spec.md:99]: Not sure that I understand this scenario and what it means...

File: packages/react-components/react-persona/Spec.md [hunk 8]
```diff
@@ -1,63 +1,279 @@
...
 ## API
 
-_List the **Props** and **Slots** proposed for the component. Ideally this would just be a link to the component's `.types.ts` file_
+**Slots**
+
+- `root`: The root slot for Persona.
+- `avatar`: The Avatar, if provided.
+- `badge`: The PresenceBadge, if provided.
+- `icon`: The icon, if provided.
+- `image`: The image, if provided.
```
Comment (layershifter) [inline packages/react-components/react-persona/Spec.md:125]: - Are these slots mutually exclusive? For example, can you use `<Persona image={foo} avatar={bar} />`?
- Are you going to bundle `Avatar` & `Image` components with `Persona`? I.e. will be there something like `state.components: { avatar: Avatar, image: Image }`?

File: packages/react-components/react-persona/Spec.md [hunk 9]
```diff
@@ -1,63 +1,279 @@
...
+- `primaryText`: Primary text, this is the only required slot. I decided to have at least this slot required since it doesn't make sense to have just the coin without the text. ⚠️ Open to feedback though.
+- `secondaryText`: Secondary text, if provided.
+- `tertiaryText`: Tertiary text, if provided.
+- `quaternaryText`: Quaternary text, if provided.
```
Comment (layershifter) [inline packages/react-components/react-persona/Spec.md:129]: `Persona` as a name is opinionated (compared even to `ItemLayout), are there reasons no not use semantic naming like `title`, `description`, etc.?

File: packages/react-components/react-persona/Spec.md (hunk 8 above)
Comment (spmonahan) [reply packages/react-components/react-persona/Spec.md:125]: Looking at the design spec there is just a space called "coin slot" where you can put an avatar, a badge, an icon, or an image. So I think they are mutually exclusive which leads me to think we should have just a "coin slot" (need a better name imo) rather than `avatar`, `badge`, `icon` and `image`.

File: packages/react-components/react-persona/Spec.md (hunk 4 above)
Comment (sopranopillow) [reply packages/react-components/react-persona/Spec.md:57]: This is a special case since you can have both the Avatar and Badge alone, having both will let you combine them.

File: packages/react-components/react-persona/Spec.md (hunk 7 above)
Comment (sopranopillow) [reply packages/react-components/react-persona/Spec.md:99]: That would be this scenario

```jsx
<Persona
  primaryText="Kevin Sturgis"
  secondaryText="Software Engineer"
  tertiaryText="Offline"
  avatar={{ name: 'Kevin Sturgis', badge: { status: 'online' } }}
  badge={{ status: 'offline', outOfOffice: true }}
/>
```

when this happens, the badge `{ status: 'offline', outOfOffice: true }` will be the one used instead of the one with `{ status: 'online' }`.

File: packages/react-components/react-persona/Spec.md (hunk 9 above)
Comment (sopranopillow) [reply packages/react-components/react-persona/Spec.md:129]: We decided to go with Persona over EntityLayout due to Entity/Item being too generic. Persona layout gives a little of context on what to expect of the component and is more familiar since v8 has the same name. For the semantic naming, there's no big reason. We could change it to semantic naming if it's preferred.

File: packages/react-components/react-persona/Spec.md (hunk 8 above)
Comment (sopranopillow) [reply packages/react-components/react-persona/Spec.md:125]: There's a problem with this. If we were to add a single slot, we would need to either modify or add types for such as slot. To be able to have the single slot, its slot definition would need to look like this:

`media: NonNullable<Slot<typeof Avatar, typeof PresenceBadge | 'img' | 'span'>>;` 

This is not possible due to AlternateAs only allowing intrinsic elements. This would mean we could not have a badge alone.

Comment (layershifter) [reply packages/react-components/react-persona/Spec.md:139]: @spmonahan I was thinking about exactly the same, but wanted to check requirements first. Do you mean following ⬇️?

```tsx
function App() {
  return (
    <>
      <Persona media={<Avatar />} />
      <Persona media={<PresenceBadge />} />
      <Persona media={<Image />} />
      <Persona media={<SomeIcon />} />
    </>
  );
}
```

@sopranopillow are there reasons why following will not work?

File: packages/react-components/react-persona/Spec.md (hunk 9 above)
Comment (layershifter) [reply packages/react-components/react-persona/Spec.md]: > We decided to go with Persona over EntityLayout due to Entity/Item being too generic.

But we use too generic `*text` slots then 😸 IMO as we went with more semantic name (`Persona`) it's better to use semantic names for slots. `description` was just an example, feel free to suggest/use own options.

File: packages/react-components/react-persona/Spec.md (hunk 7 above)
Comment (layershifter) [reply packages/react-components/react-persona/Spec.md:113]: This looks a bit strange TBH...

If you will decide to go with multiple slots (`avatar`, `image`, etc.) I would suggest to follow the order that you defined and just ignore `badge` slot if `avatar` is defined. To clarify motivation: in that snippet it looks that there will be two badges (from a slot and an avatar), but suddenly one of definitions has higher priority in others slot...

File: packages/react-components/react-persona/Spec.md (hunk 8 above)
Comment (spmonahan) [reply packages/react-components/react-persona/Spec.md:139]: That's what I was thinking. @sopranopillow points out an issue with enforcing component types with the `Slot` type helper but I wonder if that really matters? Ultimately if we have `Slot<'span'>`, as in the example, anything can be slotted and it might be nice to reduce the slot to just a span to avoid the bundle size hit of importing Avatar, PresenceBadge and Image to use Persona.

I haven't thought this through but it seems we could also type the slot with:

```ts
media: NonNullable<Slot<typeof Avatar> | Slot<typeof PresenseBadge> | Slot<typeof Image> | Slot<'span'>>
```

File: packages/react-components/react-persona/Spec.md (hunk 7 above)
Comment (sopranopillow) [reply packages/react-components/react-persona/Spec.md:113]: I see, I'm going to update the spec to follow the single slot now.

File: packages/react-components/react-persona/Spec.md (hunk 8 above)
Comment (sopranopillow) [reply packages/react-components/react-persona/Spec.md:139]: After trying out some options, I'll update the spec to have a generic `media` slot. Thanks for the feedback!

File: packages/react-components/react-persona/Spec.md (hunk 4 above)
Comment (sopranopillow) [reply packages/react-components/react-persona/Spec.md:65]: This is not relevant anymore due to the single slot change :)

**Review by marcosmoura — APPROVED** [review]

File: packages/react-components/react-persona/Spec.md (hunk 9 above)
Comment (sopranopillow) [reply packages/react-components/react-persona/Spec.md]: How about `primarySlot`, `secondarySlot`, ...? I feel like primary, secondary, and so on should be kept, but I do agree that `Text` won't work. Especially if we want to use breadcrumbs and such.

File: packages/react-components/react-persona/Spec.md [hunk 10]
```diff
@@ -1,63 +1,306 @@
...
+
+- start: Media on the left and text on the right.
+- center: Media on top and text on the bottom.
+- end: Media on the right and text on the left.
+
+There are 4 Media variants:
+
+- PersonaPresenceBadge
+- PersonaIcon
+- PersonaAvatar
```
Comment (layershifter) [inline packages/react-components/react-persona/Spec.md:114]: Can you please clarify what these components do and why they are required? For example, why I cannot do `<Persona media={<BookIcon />}>`?

Comment (sopranopillow) [reply packages/react-components/react-persona/Spec.md:114]: Done, thanks for pointing it out!

File: packages/react-components/react-persona/Spec.md [hunk 11]
```diff
@@ -1,63 +1,312 @@
...
+- Image
+
+PersonaPresenceBadge, PersonaIcon, and PersonaAvatar are needed for the sizing behavior. PersonaContextProvider is provided by Persona and then consumed by these sub-components to update the persona's media state. This media state allows Persona to correctly style the text slots based on the media's type and size. See API section for more details on the context structure.
```
Comment (layershifter) [inline packages/react-components/react-persona/Spec.md:117]: Thanks for clarifying it 👍

I am wondering if `PersonaIcon` is really needed:

```tsx
function App() {
  return (
    <>
      {/* We can't apply sizing to sized icons */}
      <Persona media={< AccessTime24Filled />} />
      {/* For unsized we can specify `fontSize` to scale them */}
      <Persona media={< AccessTimeFilled />} />
    </>
   )
}
```

WDYT about it?

---

@ling1726 AFAIR you mentioned a similar problem with `Avatar` & `Table` sizing. Did you figure out a solution?

Comment (sopranopillow) [reply packages/react-components/react-persona/Spec.md:117]: The main thing PersonaIcon does is to add a font size and let Persona know that the media is an icon. If we were to remove PersonaIcon, we then drop support of the sizing styles for icons as well. Only Avatars and PresenceBadges would have the sizing behavior. 

I think adding a disclaimer about sized icons vs non-sized icons would be better in this case. I think it's worth having the sizing functionality for the icons if possible.

Comment (layershifter) [reply packages/react-components/react-persona/Spec.md:117]: > The main thing PersonaIcon does is to add a font size and let Persona know that the media is an icon. If we were to remove PersonaIcon, we then drop support of the sizing styles for icons as well

Can we specify `fontSize` in styles for `media` slot?

Comment (sopranopillow) [reply packages/react-components/react-persona/Spec.md:117]: if that's the case, I still need to know if the media type is an icon. While I could size it with a selector like `& > svg`, I still need to know that it's an icon to size the text lines correctly.

Comment (layershifter) [reply packages/react-components/react-persona/Spec.md:118]: > if that's the case, I still need to know if the media type is an icon. While I could size it with a selector like `& > svg`, I still need to know that it's an icon to size the text lines correctly.

Sorry, but I didn't get why... Is something except avatar, image, badge and icon can be specified in `media` slot?

Comment (layershifter) [reply packages/react-components/react-persona/Spec.md:118]: > What do you think?

TBH I am a bit lost at this point and it will be easier for me to understand the implementation. Will it be okay to keep going as it is for now and then follow up on this based on the implementation?

Comment (sopranopillow) [reply packages/react-components/react-persona/Spec.md:118]: Sounds good, I'll merge this PR #24704 and make a draft PR with the implementation! Sorry for the confusion, but I agree that it'll all make sense in the Implementation.

Comment (sopranopillow) [reply packages/react-components/react-persona/Spec.md]: @layershifter one last major change, I swear 😥😥 I've updated the spec to reflect a discussion we had offline. The new trjectory of this component will be to have a strict persona that aligns with v8's persona for partners. This will ensure that we have the same persona across products. 

For the extra functionality that included images, icons and other content, we are exploring two options but leaning towards option 1:
1. We will add a recipes section to the docs and give guidance on building a media object with the components we currently have.
2. We will provide building blocks for a MediaObject and make it flexible for the user.

Thanks so much for being attentive to this PR and I really appreciate all the feedback you've provided!

File: packages/react-components/react-persona/Spec.md [hunk 12]
```diff
@@ -1,63 +1,212 @@
...
+Persona with Avatar + PresenceBadge:
+
+```jsx
+<Persona
+  avatar={{ name: 'Kevin Sturgis' }}
+  presence={{ status: 'offline', outOfOffice: true }}
+  primaryText="Kevin Sturgis"
+  secondaryText="Software Engineer"
+  tertiaryText="Offline"
+/>
```
Comment (behowell) [inline packages/react-components/react-persona/Spec.md:61]: What do you think about having the `avatar` slot be the primary slot? That would have a few benefits:

- Somewhat cleaner API since the Avatar props like `name` and `image` won't be nested inside a separate prop.
- This is how the Field controls work, and Fields conceptually do a similar thing: adding extra text and layout to an existing control.
- The value of `primaryText` could default to the `name` prop, to reduce the need to duplicate the name in multiple places.
- The `size` prop can be used to determine the text size (for the "fixed" variant). Or, the number of lines of text can be used to determine the `size` prop (for the "scaled" variant). You could even determine whether it's "fixed" or "scaled" based on whether or not the `size` prop was supplied, without relying on a separate prop.

That creates a few open questions:
1. What to do with the `presence` prop in this spec, since Avatar has a `badge` prop for the same purpose. One option would be to remove `presence` and just use `badge`. Another option is to Omit the `badge` prop from the primary slot, and instead add a `presence` slot.
2. How to pick the "presence badge" variant. One option is another prop like `badgeOnly` or `display="badge"` or something.

File: packages/react-components/react-persona/Spec.md [hunk 13]
```diff
@@ -1,63 +1,212 @@
...
+   *
+   * `fixed`: Text lines adjust to content's size.
+   * `scaled`: Content adjusts its size based on the number of text lines used.
+   */
+  sizing?: 'fixed' | 'scaled';
+
+  /**
+   * Position the Avatar/PresenceBadge will be rendered in.
+   */
+  position?: 'start' | 'center' | 'end';
```
Comment (behowell) [inline packages/react-components/react-persona/Spec.md:149]: How about `textPosition?: 'before' | 'after' | 'below'`. This aligns with other components that use `labelPosition` props.

File: packages/react-components/react-persona/Spec.md [hunk 14]
```diff
@@ -1,63 +1,212 @@
...
+  - There are no states for this component, it's a visual representation.
 - _Interaction_
   - _Keyboard_
+    - Doesn't receive focus.
   - _Cursor_
+    - Doesn't interact with cursor.
   - _Touch_
+    - Doesn't interact with touch.
   - _Screen readers_
+    - It first focuses on the media and then goes through each text line available.
```
Comment (behowell) [inline packages/react-components/react-persona/Spec.md:208]: To clarify, this is just describing the behavior of a screen reader right? Persona isn't doing anything here.

File: packages/react-components/react-persona/Spec.md [hunk 15]
```diff
@@ -1,63 +1,212 @@
...
-- Identify the **[ARIA](https://www.w3.org/TR/wai-aria-practices-1.2/) pattern** and, if the component is listed there, follow its specification as possible.
-- Identify accessibility **variants**, the `role` ([ARIA roles](https://www.w3.org/TR/wai-aria-1.1/#role_definitions)) of the component, its `slots` and `aria-*` props.
-- Describe the **keyboard navigation**: Tab Oder and Arrow Key Navigation. Describe any other keyboard **shortcuts** used
-- Specify texts for **state change announcements** - [ARIA live regions
-  ](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) (number of available items in dropdown, error messages, confirmations, ...)
-- Identify UI parts that appear on **hover or focus** and specify keyboard and screen reader interaction with them
-- List cases when **focus** needs to be **trapped** in sections of the UI (for dialogs and popups or for hierarchical navigation)
-- List cases when **focus** needs to be **moved programatically** (if parts of the UI are appearing/disappearing or other cases)
+- There's no need for `aria-*` or/and`role`. Avatar and Badge already accessible and the text labels won't need anything as well.
```
Comment (behowell) [inline packages/react-components/react-persona/Spec.md:212]: We can follow up on this later, but the Avatar's `aria-label` is the `name` prop. If that's also repeated in `primaryText`, then that name would be read out twice. Need to check that and figure out the best approach (perhaps add `aria-hidden` on the Avatar, but should definitely check into what's the right solution).

File: packages/react-components/react-persona/Spec.md (hunk 14 above)
Comment (sopranopillow) [reply packages/react-components/react-persona/Spec.md:208]: correct!

File: packages/react-components/react-persona/Spec.md (hunk 15 above)
Comment (sopranopillow) [reply packages/react-components/react-persona/Spec.md:212]: I just checked with Sarah offline, there's no correct answer to this. I checked what owa and teams are doing, and they do not hide the first line. I think we could leave as is and based on partner feedback we can refine it. One of the main reasons why I think it should stay as is, because we would have to hide it only when an Avatar is used since we can have a PresenceBadge alone. I am open to doing this if you prefer, but I do think that maybe following what we currently have is okay.

File: packages/react-components/react-persona/Spec.md (hunk 12 above)
Comment (sopranopillow) [reply packages/react-components/react-persona/Spec.md:61]: I agree, making avatar the primary slot will make things simpler.

For the open questions:
1. If we go for simplicity, letting Avatar deal with the badge would be best. Imo with the change of Avatar becoming the primary slot, it feels like presence should have its own slot. This way it would also make it easier to deal with the presence alone scenario, otherwise we would need to extract the badge from Avatar.
2. I would say badgeOnly would be the best option since I feel like display means that there will be more than two display types.

TLDR, what I think we should do is:
1. Make Avatar the primary slot and omitting its badge prop.
2. Keep the slot for presence.
3. Add `badgeOnly` prop to deal with presence variant.

What do you think?

Comment (sopranopillow) [reply packages/react-components/react-persona/Spec.md:61]: After I thought a little more about it, this could bring some problems with the slot types. If we make Avatar the primary slot, it would be NonNullable and therefore the presence only variant will have a weird API.

File: packages/react-components/react-persona/Spec.md (hunk 11 above)
Comment (layershifter) [reply packages/react-components/react-persona/Spec.md]: Sorry, but it does not seem to be aligned with the spec in Figma. Complicating usage for images/icons does not seem for like a futureproof approach 🤔 

---

IMO the latest changes also return the problem with priorities and mutually exclusive slots:

```tsx
<Persona
  avatar={{ name: 'Kevin Sturgis', badge: { status: 'online' } }}
  presence={{ status: 'offline', outOfOffice: true }}
 />
```

For me, it's the same case as `appearance` with `Button`: we don't want users to do `<Button secondary primary />`, so we have `<Button appearance="primary" />`. This follows the principle "do the right thing by default".

https://github.com/microsoft/fluentui/blob/e0a0ab1481d0fb64f2a9804376ad17e4f4f8ea01/packages/react-components/react-button/src/components/Button/Button.types.ts#L17-L27

File: packages/react-components/react-persona/Spec.md (hunk 12 above)
Comment (sopranopillow) [reply packages/react-components/react-persona/Spec.md]: @behowell quick update, I added these changes to this PR #24749. I think it works well, but let me know what you think about it!

File: packages/react-components/react-persona/Spec.md (hunk 14 above)
Comment (behowell) [reply packages/react-components/react-persona/Spec.md:219]: Hmm thinking more about this: in the `textPosition="before"` case, should the reading order follow the visual order: text then avatar? That would best be done by changing the DOM order of the slots, which is also part of my recommendation in this comment on the implementation PR: https://github.com/microsoft/fluentui/pull/24749#discussion_r971454281

File: packages/react-components/react-persona/Spec.md (hunk 15 above)
Comment (behowell) [reply packages/react-components/react-persona/Spec.md:223]: To clarify, I was suggesting possibly hiding the Avatar, not the text. We definitely shouldn't hide the text :)

File: packages/react-components/react-persona/Spec.md [hunk 16]
```diff
@@ -1,63 +1,223 @@
...
 ## Variants
 
-_Describe visual or functional variants of this control, if applicable. For example, a slider could have a 2D variant._
+> I will refer to the avatar and presence as content.
+
+There are three alignment variants:
+
+- after: Content on the left and text on the right.
+- before: Content on top and text on the bottom.
+- below: Content on the right and text on the left.
```
Comment (behowell) [inline packages/react-components/react-persona/Spec.md:72]: I think you have these swapped
```suggestion
- below: Content on top and text on the bottom.
- before: Content on the right and text on the left.
```

File: packages/react-components/react-persona/Spec.md [hunk 17]
```diff
@@ -1,63 +1,223 @@
...
+   */
+  avatar: NonNullable<Slot<typeof Avatar>>;
+
+  /**
+   * PresenceBadge to display.
+   */
+  presence: Slot<typeof PresenceBadge>;
+
+  /**
+   * Primary text to display.
```
Comment (behowell) [inline packages/react-components/react-persona/Spec.md:114]: It might be nice to have these doc comments be a little more descriptive. E.g.:
```suggestion
   * The first line of text, larger than other lines of text.
```

File: packages/react-components/react-persona/Spec.md [hunk 18]
```diff
@@ -1,63 +1,223 @@
...
+
+  /**
+   * Sizing type to use.
+   *
+   * `fixed`: Text lines adjust to content's size.
+   * `scaled`: Content adjusts its size based on the number of text lines used.
+   *
+   * @default fixed
+   */
+  sizing?: 'fixed' | 'scaled';
```
Comment (behowell) [inline packages/react-components/react-persona/Spec.md:153]: I don't think we need this `sizing` prop, since it's redundant with `size`:
* When `sizing="fixed"`, then it requires specifying the `size` prop (or rely on the default 32).
* When `sizing="scaled"`, then you can't use the `size` prop.

Therefore, we can determine which one to use just by whether or not the `size` prop was supplied.

Or, another way to describe it: the default value of the `size` prop is based on the number of lines of text (the "scaled" variant), but can be overridden by specifying the size (the "fixed" variant).

File: packages/react-components/react-persona/Spec.md [hunk 19]
```diff
@@ -1,63 +1,223 @@
...
+v8's Persona has a sub-component `PersonaCoin`, the equivalent in v9 is an Avatar. Currently Avatar is able to display a PresenceBadge, but this behavior will be deprecated in favor of Persona having this behavior.
+
+Persona has a slot for an `Avatar` and `PresenceBadge`, giving the user the ability to display an Avatar, a PresenceBadge, or Combine them into an Avatar with presence.
 
-- _Link to Open UI research_
-- _Link to comparison of v7 and v0_
-- _Link to GitHub epic issue for the converged component_
+### Text lines
+
+Other than styling and naming, the text lines in Persona remain the same.
```
[the remark this answers is not in the document]
Comment (sopranopillow) [reply packages/react-components/react-persona/Spec.md:30]: as v8, sorry I'll update that.

[the remark this answers is not in the document]
Comment (behowell) [reply packages/react-components/react-persona/Spec.md:30]: Hmm I think comparisons to v8 should be limited to the Prior Art section. Otherwise, this spec should just describe how Persona works, without using v8 as a reference.

File: packages/react-components/react-persona/Spec.md [hunk 20]
```diff
@@ -1,63 +1,213 @@
...
+  <span class="fui-Persona__quaternaryText">Quaternary Text</span>
+</div>
+```
 
-_Describe what will need to be done to upgrade from the existing implementations:_
+## Migration
 
-- _Migration from v8_
-- _Migration from v0_
+See [MIGRATION.md](./MIGRATION.md) for details.
```
Comment (behowell) [inline packages/react-components/react-persona/Spec.md:193]: It doesn't look like MIGRATION.md exists yet. Do you plan to add it in another PR?

**Review by behowell — APPROVED** [review]
Looks good to me. Thanks for making all the updates!

File: packages/react-components/react-persona/Spec.md (hunk 20 above)
Comment (sopranopillow) [reply packages/react-components/react-persona/Spec.md:193]: yes, I was planning on adding it with this issue: #24956
